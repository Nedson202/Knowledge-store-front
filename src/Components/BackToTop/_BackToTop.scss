.back-to-top {
  position: fixed;
  bottom: 20px;
  right: 30px;
  z-index: 99;
  font-size: 18px;
  border: none;
  outline: none;
	background-color: gray;
	width: 50px;
	height: 50px;
	border-radius: 50% !important;
  color: white;
  cursor: pointer;
}

.back-to-top:hover {
	box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23) !important;
}