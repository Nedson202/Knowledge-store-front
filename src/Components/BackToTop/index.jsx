import React, { Fragment } from 'react';

import { SCROLL_PARAM, SCROLL } from '../../settings';

class BackToTop extends React.PureComponent {
  state = {
    displayBackToTop: false,
  };

  componentDidMount() {
    window.addEventListener(SCROLL, this.handlePageScroll, {
      capture: true,
      passive: true
    });
  }

  componentWillUnmount() {
    window.removeEventListener(SCROLL, this.handlePageScroll, {
      capture: true,
      passive: true
    });
  }

  handlePageScroll = () => {
    let displayBackToTop = false;
    const shouldDisplayBackToTop = window.scrollY > 300;

    if (shouldDisplayBackToTop) {
      displayBackToTop = true;
    } else {
      displayBackToTop = false;
    }

    this.setState({ displayBackToTop });
  };

  scrollToTop = () => {
    window.scroll(SCROLL_PARAM);
  };

  render() {
    const { displayBackToTop } = this.state;
    return (
      <Fragment>
        {displayBackToTop && (
          <button
            className="btn btn-secondary bmd-btn-fab"
            data-testid="back-to-top"
            id="back-to-top"
            onClick={this.scrollToTop}
            type="button"
          >
            <i className="fas fa-caret-up" />
          </button>
        )}
      </Fragment>
    );
  }
}

export default BackToTop;
