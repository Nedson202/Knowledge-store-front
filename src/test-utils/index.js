import React from 'react';
import { MockedProvider } from 'react-apollo/test-utils';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import store from '../redux';

// eslint-disable-next-line react/prop-types
export const AllProviders = ({ children, customStore, apolloMocks }) => (
  <MockedProvider
    addTypename={false}
    mocks={apolloMocks}
  >
    <Provider store={customStore || store}>
      <BrowserRouter>
        {children}
      </BrowserRouter>
    </Provider>
  </MockedProvider>
);

export * from '@testing-library/react';

export const waitForTime = (waitFor = 0) => new Promise(resolve => setTimeout(resolve, waitFor));
