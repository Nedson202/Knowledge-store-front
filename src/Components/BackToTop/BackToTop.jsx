import React, { PureComponent, Fragment } from 'react';
import './_BackToTop.scss';

/* eslint-disable */
class BackToTop extends PureComponent {
  render() {
    const { backToTopMethod, displayBackToTop } = this.props;
    return (
      <Fragment>
        {displayBackToTop && (
          <button
            type="button"
            className="btn btn-secondary bmd-btn-fab back-to-top"
            onClick={backToTopMethod}
          >
            <i className="fas fa-arrow-up" />
          </button>
        )}
      </Fragment>
    );
  }
}

export default BackToTop;
