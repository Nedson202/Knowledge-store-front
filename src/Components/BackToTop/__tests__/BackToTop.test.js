import React from 'react';
import { render, fireEvent } from 'test-utils';

import BackToTop from '..';
import {
  SCROLL_301, SCROLL_0, SCROLL_500, BACK_TO_TOP
} from './constants';

window.scroll = jest.fn();

describe('BackToTop button', () => {
  it('should render', () => {
    const {
      container, asFragment, getByTestId, queryByTestId
    } = render(
      <BackToTop />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(container).toBeInTheDocument();

    expect(queryByTestId(BACK_TO_TOP)).toBeNull();
    fireEvent.scroll(window, { target: { scrollY: SCROLL_500 } });
    getByTestId(BACK_TO_TOP);
    fireEvent.scroll(window, { target: { scrollY: SCROLL_0 } });
  });

  it('should activate scroll to top', () => {
    const { getByTestId } = render(
      <BackToTop />
    );

    fireEvent.scroll(window, { target: { scrollY: SCROLL_301 } });
    const backToTopButton = getByTestId(BACK_TO_TOP);
    fireEvent.click(backToTopButton);

    window.scroll.mockClear();
  });
});
