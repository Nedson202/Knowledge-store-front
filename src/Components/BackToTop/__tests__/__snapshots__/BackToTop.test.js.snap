// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackToTop button should render 1`] = `<DocumentFragment />`;
